src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp

```diff
@@ -321,6 +375,200 @@ CHIP_ERROR EmberAttributeDataBuffer::Decode(chip::TLV::TLVReader & reader)
+    // At this point, RAW_VALUE contains the actual value, need to make it "real"
+    union
+    {
+        int64_t int_value;
+        uint64_t uint_value;
+    } value;
+
+    value.uint_value = 0;
+
+#if CHIP_CONFIG_BIG_ENDIAN_TARGET
+    bool isNegative = isSigned && (raw_bytes[0] >= 0x80);
+    if (isNegative)
+    {
+        value.int_value = -1;
+    }
+    for (int i = 0; i < static_cast<int>(byteCount); i++)
+    {
+#else
+    bool isNegative = isSigned && (raw_bytes[byteCount - 1] >= 0x80);
+    if (isNegative)
+    {
+        value.int_value = -1;
+    }
+    for (int i = static_cast<int>(byteCount) - 1; i >= 0; i--)
+    {
+#endif
+        value.uint_value <<= 8;
+        value.uint_value = (value.uint_value & ~0xFFULL) | raw_bytes[i];
+    }
+
+    if (mIsNullable && (value.uint_value == nullValue))
+    {
+        // MaxValue is used for NULL setting
+        return writer.PutNull(tag);
+    }
+
+    switch (mAttributeType)
+    {
+    case ZCL_INT8U_ATTRIBUTE_TYPE: // Unsigned 8-bit integer
+        return writer.Put(tag, static_cast<uint8_t>(value.uint_value));
+    case ZCL_INT16U_ATTRIBUTE_TYPE: // Unsigned 16-bit integer
+        return writer.Put(tag, static_cast<uint16_t>(value.uint_value));
+    case ZCL_INT24U_ATTRIBUTE_TYPE: // Unsigned 24-bit integer
+    case ZCL_INT32U_ATTRIBUTE_TYPE: // Unsigned 32-bit integer
+        return writer.Put(tag, static_cast<uint32_t>(value.uint_value));
+    case ZCL_INT40U_ATTRIBUTE_TYPE: // Unsigned 40-bit integer
+    case ZCL_INT48U_ATTRIBUTE_TYPE: // Unsigned 48-bit integer
+    case ZCL_INT56U_ATTRIBUTE_TYPE: // Signed 56-bit integer
+    case ZCL_INT64U_ATTRIBUTE_TYPE: // Signed 64-bit integer
+        return writer.Put(tag, static_cast<uint64_t>(value.uint_value));
+    case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
+        return writer.Put(tag, static_cast<int8_t>(value.int_value));
+    case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
+        return writer.Put(tag, static_cast<int16_t>(value.int_value));
+    case ZCL_INT24S_ATTRIBUTE_TYPE: // Signed 24-bit integer
+    case ZCL_INT32S_ATTRIBUTE_TYPE: // Signed 32-bit integer
+        return writer.Put(tag, static_cast<int32_t>(value.int_value));
+    default:
+        return writer.Put(tag, static_cast<int64_t>(value.int_value));
+    }
+}
+
+CHIP_ERROR EmberAttributeDataBuffer::Encode(chip::TLV::TLVWriter & writer, TLV::Tag tag) const
+{
+    EndianReader endianReader(mDataBuffer.data(), mDataBuffer.size());
+
+    switch (mAttributeType)
+    {
+    case ZCL_NO_DATA_ATTRIBUTE_TYPE: // No data
+        return writer.PutNull(tag);
+    case ZCL_BOOLEAN_ATTRIBUTE_TYPE: { // Boolean
+        uint8_t value;
+        if (!endianReader.Read8(&value).IsSuccess())
+        {
+            return endianReader.StatusCode();
+        }
+        switch (value)
+        {
+        case 0:
+        case 1:
+            return writer.PutBoolean(tag, value != 0);
+        case 0xFF:
+            return writer.PutNull(tag);
```

Comment:
This will incorrectly encode null even if `!mIsNullable`, instead of correctly erroring out.